terraform/main.tf

```diff
@@ -0,0 +1,415 @@
+    "kubernetes.io/role/internal-elb" = "1"
+  }
+}
+
+################################################################################
+# EKS Cluster - Blueprints Module
+################################################################################
+module "eks_blueprints" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.25.0"
+
+  # EKS CLUSTER
+  cluster_name       = "${local.project_name}-${local.environment}-eks"
+  cluster_version    = "1.26"
+  vpc_id             = module.vpc.vpc_id
+  public_subnet_ids  = module.vpc.public_subnets
+  private_subnet_ids = module.vpc.private_subnets
+
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
+
+  // TODO: want to test
+  // enable_irsa                     = true
+
+  // https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/619
+  node_security_group_additional_rules = {
+    # Extend node-to-node security group rules. Recommended and required for the Add-ons
+    ingress_self_all = {
+      description = "Node to node all ports/protocols"
+      protocol    = "-1"
+      from_port   = 0
+      to_port     = 0
+      type        = "ingress"
+      self        = true
+    }
+    # Recommended outbound traffic for Node groups
+    egress_all = {
+      description      = "Node all egress"
+      protocol         = "-1"
+      from_port        = 0
+      to_port          = 0
+      type             = "egress"
+      cidr_blocks      = ["0.0.0.0/0"]
+      ipv6_cidr_blocks = ["::/0"]
+    }
+    # Allows Control Plane Nodes to talk to Worker nodes on all ports. Added this to simplify the example and further avoid issues with Add-ons communication with Control plane.
+    # This can be restricted further to specific port based on the requirement for each Add-on e.g., metrics-server 4443, spark-operator 8080, karpenter 8443 etc.
+    # Change this according to your security requirements if needed
+    ingress_cluster_to_node_all_traffic = {
+      description              = "Cluster API to Nodegroup all traffic"
+      protocol                 = "-1"
+      from_port                = 0
+      to_port                  = 0
+      type                     = "ingress"
+      source_security_group_id = module.eks_blueprints.cluster_security_group_id
+    }
+  }
+
+  # List of map_users
+  map_users = [
+    {
+      userarn  = "arn:aws:iam::506657148836:user/samuel@celestia.org"
+      username = "samuel@celestia.org"
+      groups   = ["system:masters"]
+    },
+    {
+      userarn  = "arn:aws:iam::506657148836:user/viet@celestia.org"
+      username = "viet@celestia.org"
+      groups   = ["system:masters"]
+    },
+    {
+      userarn  = "arn:aws:iam::506657148836:user/alexk@celestia.org"
+      username = "alexk@celestia.org"
+      groups   = ["system:masters"]
+    },
+    {
+      userarn  = "arn:aws:iam::506657148836:user/jose@celestia.org"
+      username = "jose@celestia.org"
+      groups   = ["system:masters"]
+    }
+  ]
```

Comment:
this is not the best option to include here in the PR

Reply:
good catch! I missed it, thank you!